### ISSUE TITLE
Frequent displaying of non-errors

### ISSUE THREAD

**Issue by ebardajiz** [TypeStrong/atom-typescript#1318]
Many errors are being reported which are not errors. Hovering over the error reveals that the text that caused the error is not actually the text on screen.

An error I'm getting very frequently with code such as 

```js
import AModule from 'some/place'

AModue.method() // this line will be red-underlined indicating error
```

is `Cannot find name 'AModModModModModule'.`

The error persists and further editing of the file creates even more random errors. The plugin has to be disabled and re-enabled for it to start parsing the files correctly again.

**Comment by ericeslinger** [comment]
I'm getting this error from TSLint rather than atom-typescript, myself. The repro for me is:

1. start writing a line of code like const foo = new B
2. realize I need to import Bar, go up to the top of the file and type import { Bar } from '../stuff/bar';
3. TSLint will freeeeak out if I don't type it all at once (assuming a debounce at work here)

my theory is that the incomplete line downfile makes the whole thing not quite work. For me, I can just close the file and reopen it and it's back to normal. I only get this error when I start typing an import statement while I have some other incomplete syntax in my file.

**Comment by lierdakil** [comment]
Probably related to #1333 -- see if updating to Atom 1.19.5 fixes it for you.

**Comment by wyqydsyq** [comment]
Using `atom-1.21.0beta0` here and have been having the same issue for the last few versions.

I also often get told that an import is unused even though it is, opening and closing the file will randomly resolve this. It seems to be an issue with the plugin not reporting changes (maybe due to the new textbuffer implementation) to the linter properly, so the linter gets outdated or incomplete code to lint, returning errors that do not match what is visible on screen.

**Comment by blang** [comment]
Same on `atom-typescript-11.0.9`, `atom-1.20.1`. Occurs after tab-complete options.
Any workaround? Restarting every 2 minutes is currently my only option.

**Comment by jax-sportsbet** [comment]
Make a custom keybinding to `'typescript:clear-errors'` and hit it hard |o|

**Comment by JaxGit** [comment]
Make a custom keybinding to `'typescript:clear-errors'` and hit it..
